bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/AbstractOpenWeatherMapHandler.java

```diff
@@ -82,29 +84,50 @@ public AbstractOpenWeatherMapHandler(Thing thing) {
 
     @Override
     public void initialize() {
-        OpenWeatherMapLocationConfiguration config = getConfigAs(OpenWeatherMapLocationConfiguration.class);
-
-        boolean configValid = true;
-        if (StringUtils.trimToNull(config.getLocation()) == null) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "@text/offline.conf-error-missing-location");
-            configValid = false;
+        logger.debug("initializing handler for thing {}", getThing().getUID());
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            initializeThing(null, null);
+        } else {
+            initializeThing(bridge.getHandler(), bridge.getStatus());
         }
+    }
 
-        try {
-            location = new PointType(config.getLocation());
-        } catch (IllegalArgumentException e) {
-            location = null;
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "@text/offline.conf-error-parsing-location");
-            configValid = false;
-        }
+    private void initializeThing(@Nullable ThingHandler bridgeHandler, @Nullable ThingStatus bridgeStatus) {
+        if (bridgeHandler != null && bridgeStatus != null) {
```

Comment:
Since `bridge.getStatus()` should reflect the most recently changed status. It should be possible to refactor the logic from `bridgeStatusChanged` and `initialize` into `initializeThing`. Of course we would have to change the name here to avoid naming conflicts with our abstract method.
```suggestion
    private void initializeThingInternal() {
        Bridge bridge = getBridge();
        ThingHandler bridgeHandler = bridge != null ? bridge.getHandler() : null;
        ThingStatus bridgeStatus = bridge != null ? bridge.getStatus() : null;
        if (bridgeHandler != null && bridgeStatus != null) {
```

Reply:
In this case, you don't use at all the value provided as parameter toi bridgeStatusChanged.
I don't know how big is the risk to do that.

Reply:
The super implementation of `bridgeStatusChanged` uses that bridge status param just fine. But it doesn't take into account the bridge becoming null like your new code is doing.

As will all event handling, sometimes the event handler doesn't care about the event that is getting handled, only that something happened. This is often the case with refresh handlers which don't care what kind of event took place and thus would ignore the event parameter in its handler method.

Reply:
This is not the usual pattern I have found in different bindings.
We could at least provide the bridgeStatus as parameter to the common internal method ?